scipy/optimize/_linprog_highs.py

```diff
@@ -385,27 +423,24 @@ def _linprog_highs(lp, solver, time_limit=None, presolve=True,
     # HiGHS represents constraints as lhs/rhs, so
     # Ax + s = b => Ax = b - s
     # and we need to split up s by A_ub and A_eq
-    slack = res.get('slack', None)
-    con = None
-    if slack is not None:
-        con = slack[len(b_ub):]
-        slack = slack[:len(b_ub)]
+    if res.get('slack'):
```

Comment:
I don't really see a reason for changing this at all (why was this not clean other than not needing `None`?), so I would either change it to what @ilayn suggested or not at all.

Reply:
`'slack'` is not guaranteed to be in `res`, so what @ilayn suggested will not work as written.  This seems to me pretty close to what was suggested -- I just moved the `np.array` constructor up from below.  What changes do you not like?

Reply:
I don't understand. `'slack'` is guaranteed to be in `res` if `'slack' in res` evaluates to `True`.
I guess don't like changing to something other than what is suggested by a reviewer when the original was OK to begin with.

Reply:
Yep, this was me misunderstanding.  Sorry about that!

Reply:
It's not a big deal of course but getting a none from get and checking if the key is in dict are equivalent in Py3. Then you can branch off similarly. But again no big deal.